tests/storage/blockvolume_test.py

```diff
@@ -159,7 +159,7 @@ def test_optimal_size_cow_leaf_max(self):
     @permutations([
         # virtual_size, actual_size, optimal_size
         # Limited by max size.
-        (512 * MiB, 200 * MiB, 640 * MiB),
+        (512 * MiB, 200 * MiB, 256 * MiB),
```

Comment:
Missing test for the the case of empty source image. This should not happen since with qcow2 image empty image needs about 200k (for qcow2 header), but may happen with empty raw sparse image.

Reply:
For block raw, actual size is the virtual size, so the only missing test is empty qcow2 image.

```
$ qemu-img create -f qcow2 empty-100g.qcow2 100g
Formatting 'empty-100g.qcow2', fmt=qcow2 cluster_size=65536 extended_l2=off compression_type=zlib size=107374182400 lazy_refcounts=off refcount_bits=16

$ qemu-img check empty-100g.qcow2 
No errors were found on the image.
Image end offset: 262144
```